scripts/helpers.js

```diff
@@ -2,155 +2,155 @@
       if (link === path) itemClass += ' current';
 
       result += '<a href="' + link + '" class="' + itemClass + '">' + self.__(prefix + text) + '</a>';
-    });
-  });
+    }
+  }
 
   return result;
 });
 
 hexo.extend.helper.register('header_menu', function(className) {
-  var menu = this.site.data.menu;
-  var result = '';
-  var self = this;
-  var lang = this.page.lang;
-  var isEnglish = lang === 'en';
+  const menu = this.site.data.menu;
+  let result = '';
+  const self = this;
+  const lang = this.page.lang;
+  const isEnglish = lang === 'en';
 
-  _.each(menu, function(path, title) {
+  for (let [title, path] of Object.entries(menu)) {
     if (!isEnglish && ~localizedPath.indexOf(title)) path = lang + path;
 
-    result += '<a href="' + self.url_for(path) + '" class="' + className + '-link">' + self.__('menu.' + title) + '</a>';
-  });
+    result += `<a href="${self.url_for(path)}" class="${className}-link">${self.__('menu.' + title)}</a>`;
+  }
 
   return result;
 });
 
 hexo.extend.helper.register('canonical_url', function(lang) {
-  var path = this.page.canonical_path;
+  let path = this.page.path;
   if (lang && lang !== 'en') path = lang + '/' + path;
 
-  return this.config.url + '/' + path;
+  return full_url_for(path);
 });
 
 hexo.extend.helper.register('url_for_lang', function(path) {
-  var lang = this.page.lang;
-  var url = this.url_for(path);
+  const lang = this.page.lang;
+  let url = this.url_for(path);
 
   if (lang !== 'en' && url[0] === '/') url = '/' + lang + url;
 
   return url;
 });
 
-hexo.extend.helper.register('raw_link', function(path) {
-  return 'https://github.com/hexojs/site/edit/master/source/' + path;
-});
+hexo.extend.helper.register('raw_link', (path) => `https://github.com/hexojs/site/edit/master/source/${path}`);
 
 hexo.extend.helper.register('page_anchor', function(str) {
-  var $ = cheerio.load(str, {decodeEntities: false});
-  var headings = $('h1, h2, h3, h4, h5, h6');
+  const $ = cheerio.load(str, {decodeEntities: false});
+  const headings = $('h1, h2, h3, h4, h5, h6');
 
   if (!headings.length) return str;
 
   headings.each(function() {
-    var id = $(this).attr('id');
+    const id = $(this).attr('id');
 
     $(this)
       .addClass('article-heading')
-      .append('<a class="article-anchor" href="#' + id + '" aria-hidden="true"></a>');
+      .append(`<a class="article-anchor" href="#${id}" aria-hidden="true"></a>`);
   });
 
   return $.html();
 });
 
 hexo.extend.helper.register('lunr_index', function(data) {
-  var index = lunr(function() {
+  const index = lunr(function() {
     this.field('name', {boost: 10});
     this.field('tags', {boost: 50});
     this.field('description');
     this.ref('id');
 
-    _.sortBy(data, 'name').forEach((item, i) => {
-      this.add(_.assign({ id: i }, item));
+    data.concat().sort((a, b) => {
+      if (a.name > b.name) {
+        return 1;
+      } else if (b.name > a.name) {
+        return -1;
+      }
+      return 0;
```

Comment:
I meant these lines can be shorten to `return a.name - b.name;`, because it essentially just need to return positive/negative/zero (e.g. `-1` and `-2` are functionally similar here).

Reply:
DIrectly `a.name - b.name` will get `isNaN`.

I have post a test log above.

Reply:
ah sorry, didn't notice.